### ISSUE TITLE
Bun doesn't support tsconfig.json which consist of references to other tsconfigs

### ISSUE THREAD

**Issue by danulqua** [oven-sh/bun#4774]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

1. Create main `tsconfig.json` and another `tsconfig.app.json`
2. In the `tsconfig.app.json` add

```json
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
```

3. In the `tsconfig.json` add

```json
{
  "files": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    }
  ],
}
```

4. In the code, import something using created `@` alias

### What is the expected behavior?

Imports using alias work as expected.

### What do you see instead?

"Cannot find module" error occurs:

<img width="1440" alt="image" src="https://github.com/oven-sh/bun/assets/38916225/0835add3-541e-4a51-8b80-68e4f2879336">

As soon as you add `paths` in the main `tsconfig.json`, it works as expected:

<img width="1440" alt="image" src="https://github.com/oven-sh/bun/assets/38916225/23ce04a3-8952-4dd2-83eb-2648ad5adec9">

(Don't mind other errors, I'm playing around with an existing Vue project and it seems that Bun isn't able to work with it).

### Additional information

In conclusion, I guess that Bun doesn't take into account other tsconfigs referenced by the main tsconfig.

**Comment by zavodnyuk** [comment]
I'm working on monorepo and also have multiple tsconfig with root config and others are referencing to it. Facing same issue that paths aren't considered and ending up with error that Bun can't resolve `@someAlias/dir/file`

**Comment by Madscientiste** [comment]
Same as @zavodnyuk, i have multiples tsconfig references.

is there any workaround ?

**Comment by Jarred-Sumner** [comment]
bun supports `extends` as a relative path

**Comment by nickcherry** [comment]
In case it wasn't clear from @Jarred-Sumner's comment above, it seems that `extends` must be a string, not an array of strings.

**Comment by mrhyde** [comment]
> bun supports `extends` as a relative path

Any plans to support npm packages in the `extends` field?

**Comment by dilame** [comment]
I'm also working on monorepo with references, would be nice to have support for it

**Comment by Pedromigacz** [comment]
I just created an issue [14694](https://github.com/oven-sh/bun/issues/14694). I think it's related but I'm not sure it's a duplicate as my tsconfig.json seems to work fine except the path property.

**Comment by mnpenner** [comment]
I have an app that uses one `jsxImportSource` for the client and a different `jsxImportSource` for the server. Bun won't respect the `jsxImportSource` in `server/tsconfig.json`, it only reads from the root `tsconfig.json`. Very frustrating.

I can't even trick it with

```
bun --tsconfig-override=server/tsconfig.json run --hot server/serve.ts
```

It just ignores it, for whatever reason.

**Comment by rodrigoclp** [comment]
Hello guys!

Anything new about this issue?

TSC always return errors because cannot find modules in a monorepo structure with multiple tsconfig files.

**Comment by dortiz-prsg** [comment]
Just ran into this, thought I was crazy... even started a new project from scratch.

**Comment by EKliptus** [comment]
I am facing a similar issue bun not with tsconfig directly... Please tell me if I need to create another issue or if I misunderstand anything...

```bash
├── base/src/configs/cmsConfig.ts
├── sites/demo
│   └── tsconfig.json
├── scripts
│   └── my-script.ts
└── tsconfig.json
```
In the main tsconfig in root :
```jsonc
"compilerOptions": {
    "baseUrl": ".",
    "paths": {
          "@base/*": ["base/src/*"],
    }
} 
```
In the sites/xxx/tsconfig.json :
```jsonc
"compilerOptions": {
    "baseUrl": ".",
    "paths": {
          "@base/*": ["../../base/src/*"],
    }
} 
```
base/src/configs/cmsConfig.ts :
```ts
// ALIAS NOT RESOLVE correctly in my-script.ts but ok in sites projects (nedded for sites/xxx , not for my-script.ts )
import { type Cms, CmsDistType } from "@base/types/cmsConfigs";
// This is the needed const for my-script.ts !!! :
export const GLOBAL_VERSION = config.sveltiaCmsVersion;
// others nedded for sites/xxx only
```
For `my-script.ts`
I have error : `Cannot find module '@base/types/cmsConfigs' from 'E:\dev-env\my-project\base\src\configs\cmsConfig.ts'`

imports in `my-script.ts` :
`import { GLOBAL_VERSION } from "@base/configs/cmsConfig";`
or
`import { GLOBAL_VERSION } from "../../base/src/configs/cmsConfig";`
correctly resolve the file...
**BUT it is the second import alias from `@base` which is not relative to the subproject tsconfig !?**

```bash
my-script.ts
  └─[import form `@base` (root tsconfig.json)]─> cmsConfig 
                                                   └─[import form `@base` (project tsconfig.json)]
# Fail because not use sub-project tsconfig but refer only to the root tsconfig !?
```

Thanks !

**Comment by NoodleSoup** [comment]
Running against this issue as well with a Nuxt 3 project. Trying to run this with project with `bun --bun run dev` gives me a lot of import errors. Hoping to see this fixed as Nuxt generates a tsconfig file for all it's imports located in `.nuxt/tsconfig.json`.

**Comment by spamshaker** [comment]
The project references support is crucial as fetch api differs for browser/bun also for Blobs and (window with all DOM of course :)) thus there should be a way to keep them separated

**Comment by pleclech** [comment]
Also got bitten by that with Bun 1.3
